plugins/nodes/src/js/components/NodesTable-new.tsx

```diff
@@ -0,0 +1,311 @@
+  cpuSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableCPUColumn";
+import {
+  membarRenderer,
+  membarSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableMemBarColumn";
+import {
+  memSorter,
+  memRenderer,
+  memSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableMemColumn";
+import {
+  diskbarRenderer,
+  diskbarSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableDiskBarColumn";
+import {
+  diskSorter,
+  diskRenderer,
+  diskSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableDiskColumn";
+import {
+  gpubarSizer,
+  gpubarRenderer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableGPUBarColumn";
+import {
+  gpuSorter,
+  gpuRenderer,
+  gpuSizer
+} from "#PLUGINS/nodes/src/js/columns/NodesTableGPUColumn";
+
+interface NodesTableProps {
+  hosts: NodesList;
+  nodeHealthResponse: boolean;
+  masterRegion: string;
+}
+
+interface NodesTableState {
+  data: Node[];
+  sortDirection: SortDirection;
+  sortColumn: string;
+}
+
+// This whole component (except `render()`) will basically go into data layer.
+// After we moved to DL, `NodesTable` component should be stateless/functional.
+export default class NodesTable extends React.Component<
+  NodesTableProps,
+  NodesTableState
+> {
+  constructor() {
+    super();
+
+    this.state = {
+      data: [],
+      sortColumn: "hostname",
+      sortDirection: "ASC"
+    };
+
+    this.handleSortClick = this.handleSortClick.bind(this);
+  }
+
+  sortData(data: Node[], sortColumn: string): Node[] {
+    switch (sortColumn) {
+      case "hostname":
+        return hostnameSorter(data);
+      case "region":
+        return regionSorter(data);
+      case "zone":
+        return zoneSorter(data);
+      case "health":
+        return healthSorter(data);
+      case "tasks":
+        return tasksSorter(data);
+      case "cpu":
+        return cpuSorter(data);
+      case "mem":
+        return memSorter(data);
+      case "disk":
+        return diskSorter(data);
+      case "gpu":
+        return gpuSorter(data);
+      default:
+        return data;
+    }
+  }
+
+  handleSortClick(columnName: string): void {
+    if (columnName === this.state.sortColumn) {
+      this.setState({
+        sortDirection: this.state.sortDirection === "ASC" ? "DESC" : "ASC",
+        data: this.state.data.slice().reverse()
+      });
+    } else {
+      this.setState({
+        sortColumn: columnName,
+        sortDirection: "ASC",
+        data: this.sortData(this.state.data.slice(), columnName)
+      });
+    }
+  }
+
+  componentWillReceiveProps(nextProps: NodesTableProps): void {
+    const sortedData = this.sortData(
+      nextProps.hosts.getItems().slice(),
+      this.state.sortColumn
+    );
+    this.setState({
+      data:
+        this.state.sortDirection === "ASC" ? sortedData : sortedData.reverse()
+    });
+  }
+
+  render() {
+    const { data, sortColumn, sortDirection } = this.state;
+
+    // With React 16.4 we can use createContext() instead of passing down props
```

Comment:
React 15 also has a context API, so I see no use in deferring this

Reply:
Yup, but I decided to defer that because UI Kit uses react 16.4 already and the `SortableColumnHeader` will go there, also I think we want to think about this `tableContext` a bit more instead of just throwing it in here.

Reply:
If we want to think about it a bit more, can we remove the comment? It appears to me that react 16 is the only thing holding us back here, when in fact we need to understand if we want to have a context and how it is structured

Reply:
ok, I'll change the comment and link a JIRA for the context thingy